Tests/Data/Parabolic/T/3D_3BHEs_array/bcs_tespy.py

```diff
@@ -0,0 +1,192 @@
+###
+#Copyright(c) 2012 - 2019, OpenGeoSys Community(http:  // www.opengeosys.org)
+#Distributed under a Modified BSD License.
+#See accompanying file LICENSE.txt or
+#http:  // www.opengeosys.org/project/license
+###
+
+import sys
+print(sys.version)
+import os
+import numpy as np
+import pandas as pd
+import OpenGeoSys
+from tespy import  cmp, con, nwk, hlp, cmp_char
+from tespy import nwkr
+
+###User setting++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#parameters
+#refrigerant parameters
+refrig_density = 992.92 #kg/m3
+#% % switch for special boundary conditions
+switch_dyn_demand = 'on'# 'on','off', switch of the function for dynamic thermal demand from consumer
+switch_dyn_frate = 'off'# 'on','off', switch of the function for dynamic flowrate in BHE
+
+#% % timecurve setting
+def timerange(t):
+#month for closed network
+    timerange_nw_off_month = [-9999]#No month for closed network
+    nw_status = 'on'
+    t_trans = int((t-1)/86400/30) + 1#t-1 to avoid the calculation problem at special time point,e.g. t = 2592000.
+    t_trans_month = t_trans
+    if t_trans_month > 12:
+        t_trans_month = t_trans - 12*(int(t_trans/12))
+    if t_trans_month in timerange_nw_off_month:
+        nw_status = 'off'
+    return t_trans, t_trans_month, nw_status
+
+#% % consumer thermal load
+#month demand
+def consumer_demand(t):#dynamic thermal demand from consumer
+#thermal demand in each month (assumed specific heat extraction rate* lenth of BHE* number of BHE)
+    month_demand = [-25*50*3,-25*50*3,-25*50*3,-25*50*3,-25*50*3,-25*50*3,-25*50*3,-25*50*3,-25*50*3,-25*50*3,-25*50*3,-25*50*3]
+    return month_demand[t -1]
+
+#% % dynamic hydraulic flow rate
+#month demand
+def dyn_frate(t):#dynamic flowrate in BHE
+#flow rate in kg / s time curve in month
+    month_frate = [-9999]
+    return month_frate[t -1]
+###End User setting+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+#% % create network dataframe
+def create_dataframe():
+#return dataframe
+    df_nw = pd.read_csv('./pre/bhe_network.csv', delimiter=';',
```

Comment:
Importing whole pandas only for reading csv looks not good.
Wouldn't simple csv module be sufficient? Or at least only the read function from pandas?

Reply:
Instead of the entire Pandas lib, we import only the cvsreader instead.

Reply:
In the bcs_tespy.py, we are using Pandas module not only for the csv_read function, also the Dataframe type data structure. For instance, in the following code: 
df.loc[df.index[i], 'flowrate'] =... (line #90 in the  bcs_tespy.py)
Here, we need to use the "loc" attribute from the Pandas.Dataframe to locate the related data entry in the array. In numpy module there is no "loc" attribute, therefore we need to change the data structure entirely in the script if only numpy module is allowed. Therefore,  we prefer to import the Pandas.read_csv package.